midend/lib/Conversion/LowerGemmini/LowerGemminiPass.cpp

```diff
@@ -152,6 +152,95 @@ class PrintOpLowering : public ConversionPattern {
   }
 };
 
+class PrintScalarOpLowering : public ConversionPattern {
+public:
+  explicit PrintScalarOpLowering(MLIRContext *context)
+      : ConversionPattern(gemmini::PrintScalarOp::getOperationName(), 1,
+                          context) {}
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto context = rewriter.getContext();
+    auto loc = op->getLoc();
+
+    ModuleOp parentModule = op->getParentOfType<ModuleOp>();
+
+    auto printfRef = getOrInsertPrintf(rewriter, parentModule);
```

Comment:
Do not use `auto`,The data types here can be confusing for people who are not familiar with this code.